sphinx/config.py

```diff
@@ -278,9 +280,17 @@ class Config:
     def __init__(self, config: dict[str, Any] | None = None,
                  overrides: dict[str, Any] | None = None) -> None:
         raw_config: dict[str, Any] = config or {}
+        constants = raw_config.pop('__constants__', None) or frozenset()
```

Comment:
If someone is using this name for something else, it's a bad idea to actually change it IMO.